R/stats.R

```diff
@@ -22,23 +22,34 @@ n_complete <- function(x) {
   length(x) - n_missing(x)
 }
 
+
+#' Create a contingency table and arrange its levels in descending order
+#' 
+#' @param x An object that can be interpreted as a factor (including logical)
+#' @return A "table" object, which will be treated as a named numeric vector
+#' @export
+
+sorted_count <- function(x) {
+  tab <- table(x, useNA = "always")
+  out <- setNames(as.integer(tab), names(tab))
+  sort(out, decreasing = TRUE)
```

Comment:
You'd need separate treatment for ordered factors that doesn't do this sorting.

Reply:
We're sorting the counts, which is separate from the ordering of the factor itself.

Imagine that we're summarizing a table with a factor with 100 levels. The previous could function would try and show counts for every level. That's not helpful, and it goes against the "skim" concept of the package.

Now, we just take the most popular levels and show them in the results. The formatting options .levels controls how many these to show.

Reply:
I agree about limiting the number of categories in both ordered and regular factors, that's why in the branch they are limited to 45 characters.  But ordering in ordered factors has real meaning, I don't think it makes sense to say small, large, medium, xsmall, xxl.   I think right now we are not doing anything at all to indicated that a factor is ordered or not either so in terms of skimming that's actually a problem to consider if we are going to throw them in with the regular factors. That's information you would want to know in the data set.  Maybe we can add something when creating this string?